deepchem/data/datasets.py

```diff
@@ -96,6 +96,8 @@ def pad_batch(batch_size, X_b, y_b, w_b, ids_b):
 
     # Fill in batch arrays
     start = 0
+    # Only the first set of copy will be counted in training loss
+    w_out[start:start + num_samples] = w_b[:]
```

Comment:
I think basically this iterbatches function should work for all kinds of dataset structures(different featurizers generate different X_b), which makes it hard to have a uniform way of padding. The change on w_b should be enough for the loss to be correctly mapped, so I am personally satisfied with the structure now.